### PR TITLE
Fix bug with msaad52v1 only admins consent to apps

### LINKED ISSUE #1022: Bug with MS.AAD.5.2v1 - "Only administrators SHALL be allowed to consent to applications."
# 🐛 Summary #

What's wrong? Please be specific.

In the report, MS.AAD.5.2v1 is displaying "Fail", however, I've verified in Azure Portal --> Enterprise Apps --> Consent & Permissions, that "User consent for applications" is set to "Do not allow user consent". I've flipped it on and off, and ran the report again, but it still fails with: 

1 authorization policies found that allow non-admin users to consent to third-party applications:
authorizationPolicy

Invoke-SCuBA -ProductNames aad

## Expected behavior ##

I expected that 5.2 would pass.

## Any helpful log output or screenshots ##

Paste the results here:

```console

```

Add any screenshots of the problem here.

### PR DESCRIPTION
**Pull request by tkol2022**
#  #
  

## 🗣 Description ##

Fixed the bug in AAD policy 5.2 user consent to apps, that was reported by multiple users. Microsoft added some data elements to the respective JSON array that broke the Rego logic, so I revised the logic, updated the unit tests to match the new state and updated the functional tests. I also made a minor fix to the RunUnitTests.ps1 to correct a flaw when passing the "-c" control group parameter. 

Closes #1022 

## 🧪 Testing ##

Tested against E5, G5, G3 and GCC High.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] PR targets the correct parent branch (e.g., main or release-name) for merge.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] Changes are sized such that they do not touch excessive number of files.
- [x] *All* future TODOs are captured in issues, which are referenced in code comments.
- [x] These code changes follow the ScubaGear [content style guide](https://github.com/cisagov/ScubaGear/blob/main/CONTENTSTYLEGUIDE.md).
- [x] Related issues these changes resolve are linked preferably via [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [x] All relevant type-of-change labels added.
- [x] All relevant project fields are set.
- [x] All relevant repo and/or project documentation updated to reflect these changes.
- [x] Unit tests added/updated to cover PowerShell and Rego changes.
- [x] Functional tests added/updated to cover PowerShell and Rego changes.
- [x] All relevant functional tests passed.
- [ ] All automated checks (e.g., linting, static analysis, unit/smoke tests) passed.

## ✅ Pre-merge checklist ##

- [ ] PR passed smoke test check.
- [ ] Feature branch has been rebased against changes from parent branch, as needed

  Use `Rebase branch` button below or use [this](https://www.digitalocean.com/community/tutorials/how-to-rebase-and-update-a-pull-request) reference to rebase from the command line.
- [ ] Resolved all merge conflicts on branch
- [ ] Notified merge coordinator that PR is ready for merge via comment mention

## ✅ Post-merge checklist ##

- [ ] Feature branch deleted after merge to clean up repository.
- [ ] Verified that all checks pass on parent branch (e.g., main or release-name) after merge.

### CODE REVIEW

File: Testing/RunUnitTests.ps1 [hunk 1]
```diff
@@ -156,7 +156,7 @@ function Invoke-ControlGroupItem {
 
             elseif(Test-Path -Path $Filename.Fullname -PathType Leaf) {
                 Write-Output "`nTesting Control Group $ControlGroup"
-                ..\opa_windows_amd64.exe test $RegoPolicyPath .\$($Filename.Fullname) $Flag
+                & $OPAExe test $RegoPolicyPath .\$($Filename.Fullname) $Flag
```
Comment (Sloane4) [inline Testing/RunUnitTests.ps1:159]: This conflicts with the AAD PR. Either add the entire fix to this PR or remove to prevent merge conflicts.

**Review by Sloane4 — APPROVED** [review]

**Review by nanda-katikaneni — APPROVED** [review]
Tested against E5, G5, G3 and gcchigh tenants for both positive and negative cases - works as expected. The baseline instructions need a slight update (opened a separate issue for that). Approved.

File: PowerShell/ScubaGear/Rego/AADConfig.rego [hunk 2]
```diff
@@ -520,22 +520,26 @@ tests contains {
...
     some Policy in input.authorization_policies
-    count(Policy.PermissionGrantPolicyIdsAssignedToDefaultUserRole) != 0
+    "ManagePermissionGrantsForSelf.microsoft-user-default-legacy" in Policy.PermissionGrantPolicyIdsAssignedToDefaultUserRole
 }
 
-# Get all policy Ids
+BadDefaultGrantPolicies contains Policy.Id if {
+    some Policy in input.authorization_policies
+    "ManagePermissionGrantsForSelf.microsoft-user-default-low" in Policy.PermissionGrantPolicyIdsAssignedToDefaultUserRole
+}
```
[the remark this answers is not in the document]
Comment (tkol2022) [reply PowerShell/ScubaGear/Rego/AADConfig.rego:532]: Me 2.

File: Testing/RunUnitTests.ps1 (hunk 1 above)
Comment (tkol2022) [reply Testing/RunUnitTests.ps1:159]: This PR is going to be merged first and the code change is the same as the other PR anyway.

**Comment by tkol2022** [comment]
@nanda-katikaneni ready for merge

**Pull request merged**